src/StreamJsonRpc/Reflection/MessageFormatterProgressTracker.cs

```diff
@@ -0,0 +1,168 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace StreamJsonRpc.Reflection
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Reflection;
+    using Microsoft;
+    using Microsoft.VisualStudio.Threading;
+    using Newtonsoft.Json.Linq;
+
+    /// <summary>
+    /// Class containing useful methods to help message formatters implement support for <see cref="IProgress{T}"/>.
+    /// </summary>
+    public class MessageFormatterProgressTracker
+    {
+        /// <summary>
+        /// Dictionary used to map progress id token to its corresponding <see cref="ProgressParamInformation" /> instance containing the progress object and the necessary fields to report the results.
+        /// </summary>
+        public readonly Dictionary<long, ProgressParamInformation> ProgressMap = new Dictionary<long, ProgressParamInformation>();
```

Comment:
>public [](start = 8, length = 6)

Given I don't think we want callers to *clear* our dictionary, but really just add and remove (and we already have a method on this class that adds) can we make this private and add a method to this class to remove as well, if callers need to do that?

Reply:
Fixed in latest update.